src/Razor/src/Microsoft.NET.Sdk.Razor/GenerateAspNetCoreStaticAssetsManifest.cs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class GenerateAspNetCoreStaticAssetsManifest : Task
+    {
+        private const string ContentRoot = nameof(ContentRoot);
+        private const string BasePath = nameof(BasePath);
+        
+        [Required]
+        public string TargetManifestPath { get; set; }
+
+        [Required]
+        public ITaskItem[] ContentRootDefinitions { get; set; }
+
+        public override bool Execute()
+        {
+            if(!ValidateArguments())
+            {
+                return false;
+            }
+
+            return ExecuteCore();
+        }
+
+        private bool ExecuteCore()
+        {
+            var document = new XDocument(new XDeclaration("1.0", "utf-8", "yes"));
+            var root = new XElement(
+                "AspNetCoreStaticAssets",
```

Comment:
What would you think about dropping the name `AspNetCore` from a lot of these names (targets, task, this)? It just seems jarring to me and doesn't really add anything descriptive.

Reply:
Maybe. I added it to have some sort of namespace for msbuild. Would we be ok if te replace it with something like WebAssets or similar?

Totally open to suggestions here.

Reply:
A few thoughts on naming:
"WebAssets"
"ReferencedWebAssets"
"ReferencedWebContent"

@SteveSandersonMS chip in if you like.

Reply:
I'm not toally convinced we should have a *namespace*. We are building **the** feature for static assets.

Reply:
@rynowak I'm not confortable with StaticAssets by itself, I would prefer StaticWebAssets or soemthing like that, that hinted a bit more that this is meant for static content you serve from a web.

Reply:
I'd be happy with StaticWebAssets

Reply:
StaticWebAssets sounds good.